lib/clamav/client.rb

```diff
@@ -49,5 +49,18 @@ def resolve_default_socket
         ::UNIXSocket.new(unix_socket || '/var/run/clamav/clamd.ctl')
       end
     end
+
+    def ping
+      execute Commands::PingCommand.new
+    end
+
+    def scan(file)
+      execute Commands::InstreamCommand.new(file) if file.class = StringIO
```

Comment:
`file.class = StringIO` can't work as `#class=` isn't a method on any known class in Ruby's API.  I think that what you want to test is equality between that file's type and `StringIO`.  This could be done using `is_a?`.

Also, chaining calls to `execute` won't make `scan` return the right value (if `file` is indeed a StringIO, you'll still end up executing the line below, and the method will return whatever the line below returns.

You could want to use `if ... else` here.